core/src/mender-client.c

```diff
@@ -1111,9 +1092,13 @@ mender_client_update_work_function(void) {
                             char       *new_provides  = NULL;
                             const char *artifact_name = NULL;
                             if (MENDER_OK == (ret = mender_prepare_new_provides(mender_artifact_ctx, &new_provides, &artifact_name))) {
-                                cJSON_AddStringToObject(mender_client_deployment_data, "provides", new_provides);
+                                if (MENDER_OK != (ret = mender_deployment_data_set_provides(mender_client_deployment_data, new_provides))) {
+                                    mender_log_error("Failed to set deployment data provides");
+                                }
                                 /* Replace artifact_name with the one from provides */
-                                cJSON_ReplaceItemInObject(mender_client_deployment_data, "artifact_name", cJSON_CreateString(artifact_name));
+                                if (MENDER_OK != (ret = mender_deployment_data_set_artifact_name(mender_client_deployment_data, artifact_name))) {
```

Comment:
This potentially overrides a previous failure. I think we only want to do this if the above succeeds.